cluster/kubernetes/images.go

```diff
@@ -140,5 +141,16 @@ func (c *Cluster) ImagesToFetch() registry.ImageCreds {
 		}
 	}
 
+	// remove images based on the glob exclusion list
```

Comment:
I wonder if it is possible to exit (continue) early if an image it be excluded, rather than do all the work then undo some of it. The way `mergeCredentials` works is to get all the secrets, then record them against each images used by the workload. Exiting early would mean testing whether _any_ image in the workload is to be included, and only continuing if that's the case. In other words, compile the list of images in a workload that are to be included, first, and if len==0, exit.

This adds a bit of complication to `mergeCredentials`, but not much more than it removes from here; with the benefit that it'll reduce some Kubernetes API use (no use looking up image secrets for images we don't care about).

If it turns out that it's much simpler to do it here, I'd consider putting exclusion in a separate ImageFetch wrapper, like the ECR auth bit, instead.

Reply:
In any case, it'd be good to see some tests of the exclusion -- both as a guard against regressions, and as docs of how it's intended to work.

Reply:
I've moved the exclusion logic to `mergeCredentials` and added tests.